controllers/extendeddaemonsetreplicaset/filters.go

```diff
@@ -101,7 +106,11 @@ func FilterAndMapPodsByNode(logger logr.Logger, replicaset *datadoghqv1alpha1.Ex
 
 	// add duplicated pods to the pod deletion slice
 	for _, pod := range duplicatedPods {
-		logger.V(1).Info("PodToDelete", "reason", "duplicatedPod", "pod.Name", pod.Name, "node.Name", pod.Spec.NodeName)
+		nodeName, err := podutils.GetNodeNameFromPod(pod)
+		if err != nil {
+			continue
+		}
+		logger.V(1).Info("PodToDelete", "reason", "duplicatedPod", "pod.Name", pod.Name, "node.Name", nodeName)
```

Comment:
it looks like if we don't find the `nodeName`, we skip the log line but still delete the pod. is that intended?

Reply:
right, 
I think, this case should not happen, and it is fine if we delete it in that case.
I will update the code to remove the error check and print the log line in any case.